hazelcast/src/hazelcast/client/internal/nearcache/impl/KeyStateMarkerImpl.cpp

```diff
@@ -22,10 +22,15 @@ namespace hazelcast {
         namespace internal {
             namespace nearcache {
                 namespace impl {
-                    KeyStateMarkerImpl::KeyStateMarkerImpl(int count) : markCount(count), marks(count) {
+                    KeyStateMarkerImpl::KeyStateMarkerImpl(int count) : markCount(count) {
+                        marks = new util::Atomic<int32_t>[count];
+                        for (int i = 0; i < count; ++i) {
+                            marks[i] = 0;
+                        }
                     }
 
                     KeyStateMarkerImpl::~KeyStateMarkerImpl() {
+                        delete [] marks;
```

Comment:
auto_ptr for marks ?

Reply:
I don't think auto_ptr works for this case. I changed with boost::scoped_array for RAII. Since scoped_array is used internally, it does not seem to be released to user.